### PR TITLE
ginkgo: add v1.9.0

### PR DESCRIPTION
**Pull request by MarcelKoch**

### CODE REVIEW

File: var/spack/repos/builtin/packages/ginkgo/package.py [hunk 1]
```diff
@@ -25,7 +25,9 @@ class Ginkgo(CMakePackage, CudaPackage, ROCmPackage):
     license("BSD-3-Clause")
 
     version("develop", branch="develop")
-    version("master", branch="master")
+    version("main", branch="main")
+    version("master", branch="master", deprecated=True)
+    version("1.9.0", branch="20cfd68795f58078898da9890baa311b46845a8b")  # v1.9.0
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/ginkgo/package.py]:
```suggestion
    version("1.9.0", commit="20cfd68795f58078898da9890baa311b46845a8b")  # v1.9.0
```

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Confirmed the branches exist. Please use the `commit` argument for the `1.9.0` hash.

File: var/spack/repos/builtin/packages/ginkgo/package.py [hunk 2]
```diff
@@ -53,6 +55,7 @@ class Ginkgo(CMakePackage, CudaPackage, ROCmPackage):
     variant("hwloc", default=False, description="Enable HWLOC support")
     variant("sde", default=False, description="Enable PAPI SDE support", when="@1.7.0:")
     variant("mpi", default=False, description="Enable MPI support", when="@1.5.0:")
+    variant("half_precision", default=True, description="Enable half-precision support")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/ginkgo/package.py]: Did you intend to use this in some way?  At this point the variant has no affect on the build.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Thank you for the update.

Confirmed the branches and commit but do not see how the new variant is being used in the package. If it isn't going to affect the build, the new variant should be removed.

File: var/spack/repos/builtin/packages/ginkgo/package.py (hunk 2 above)
Comment (MarcelKoch) [reply var/spack/repos/builtin/packages/ginkgo/package.py]: Yes, I added it now as a CMake option.

**Comment by tldahlgren** [comment]
One stack exceeded time limit for generation. Trying again:

@spackbot run pipeline

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**